app/code/Magento/StoreGraphQl/Model/Resolver/Store/Identity.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\StoreGraphQl\Model\Resolver\Store;
+
+use Magento\Framework\GraphQl\Query\Resolver\IdentityInterface;
+use Magento\Framework\App\Config;
+
+class Identity implements IdentityInterface
+{
+    /**
+     * @var string
+     */
+    private $cacheTag = Config::CACHE_TAG;
+
+    /**
+     * @inheritDoc
+     */
+    public function getIdentities(array $resolvedData): array
+    {
+        $ids =  empty($resolvedData) ?
+            [] : array_merge([$this->cacheTag], array_map(function ($key) {
+                return sprintf('%s_%s', $this->cacheTag, $key);
+            }, array_keys($resolvedData)));
+        return $ids;
```

Comment:
Is this way of setting the $ids really necessary? Seems like there are too many tags?

Since `Magento\StoreGraphQl\Model\Resolver\StoreConfigResolver::resolve()` output does not depend on the query fields, but on the store picked up from the context, as an idea I would suggest that the tag ids could be

- one fixed, e.g. `store_config`
- one variable, that contains the store id, e.g. 'store_config_5' (in the case where the current store id is 5)

Reply:
I agree with @pmarjan. this generates so many tags which we can't purge and they're general for all stores.

usually if config is invalidated for store 1 then CONFIG_1 is purged. you don't purge each field, because we don't have per field cache, if the value of some field changes in the admin for that store, all store config for that store needs to be purged so CONFIG_1 needs to be purged.

SImilarly if we modify a config setting that doesn't have a corespondent in a site (like modifying something in Default config  ) then we purge CONFIG tag for all sites

This behavior doesn't happen now. We don't purge nor CONFIG nor CONFIG_1 when we change something in config.

For this reason if we are to implement this feature, we do need to add the purging functionality
It's how caching works